### ISSUE TITLE
Doesn't like TypeVar now in 2.13.3 - C0103: Type variable name....

### ISSUE THREAD

**Issue by stevie-b-g** [PyCQA/pylint#6050]
### Bug description

pylint 2.13.3 is now compliant about C0103: Type variable name when using TypeVar

https://www.pythonsheets.com/notes/python-typing.html#using-typevar-and-generic-as-class-template

### Configuration

```ini
from typing import TypeVar

CacheType = TypeVar("CacheType")
```

### Command used

```shell
─○ cat a.py

   1   │ from typing import TypeVar
   2   │
   3   │ CacheType = TypeVar("CacheType")

╰─○ pylint a.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

╰─○ pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.9.6 (default, Aug 31 2021, 14:55:02)
[Clang 12.0.5 (clang-1205.0.22.11)]
╰─○ pip install --upgrade pylint
...
...
╰─○ pylint --version
pylint 2.13.3
astroid 2.11.2
Python 3.9.6 (default, Aug 31 2021, 14:55:02)
[Clang 12.0.5 (clang-1205.0.22.11)]
╰─○ pylint a.py
************* Module a
a.py:3:0: C0103: Type variable name "CacheType" doesn't conform to predefined naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 10.00/10, -5.00)

╰─○ m
```

### Pylint output

```shell
a.py:3:0: C0103: Type variable name "CacheType" doesn't conform to predefined naming style
```

### Expected behavior

Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.13.3
astroid 2.11.2
Python 3.9.6 (default, Aug 31 2021, 14:55:02)
```

**Comment by jacobtylerwalls** [comment]
Thanks for reporting. FYI this was an intentional change, see discussion at #6003.
